sdk/compiler/daml-extension/src/extension.ts

```diff
@@ -318,10 +332,28 @@ function getLanguageServerArgs(
   return serverArgs;
 }
 
-export function createLanguageClient(
+async function getSdkVersion(
+  damlPath: string,
+  projectPath: string | undefined,
+): Promise<string | undefined> {
+  // Ordered by priority
+  const selectionStrings = [
+    "selected by env var ",
+    "project SDK version from daml.yaml",
+    "default SDK version for new projects",
+  ];
+  const { stdout } = await exec(damlPath + " version", { cwd: projectPath });
+  const lines = stdout.split("\n");
+  for (const selection of selectionStrings) {
+    const line = lines.find((line: string) => line.includes(selection));
+    if (line) return line.trim().split(" ")[0];
+  }
```

Comment:
Considered using something like `daml exec env` or `daml exec printenv DAML_SDK_VERSION`, but the windows equivalents are messy